custodial-copy-backend/src/main/scala/uk/gov/nationalarchives/custodialcopy/Main.scala

```diff
@@ -40,13 +40,15 @@ object Main extends IOApp {
       deleted <- c.downField("deleted").as[Boolean]
     } yield {
       val typeAndRef = id.split(":")
-      val ref = UUID.fromString(typeAndRef.last)
-      val entityType = typeAndRef.head
-      entityType match {
-        case "io" => IoReceivedSnsMessage(ref, deleted)
-        case "co" => CoReceivedSnsMessage(ref, deleted)
-        case "so" => SoReceivedSnsMessage(ref, deleted)
-      }
+      typeAndRef match
+        case Array(entityType, refString) =>
+          val ref = UUID.fromString(refString)
+          entityType match {
+            case "io" => IoReceivedSnsMessage(ref)
+            case "co" => CoReceivedSnsMessage(ref)
+            case "so" => SoReceivedSnsMessage(ref)
+          }
+        case Array(ref) => DeletedReceivedSnsMessage(UUID.fromString(ref))
```

Comment:
1. Maybe `DeletionReceivedSnsMessage`?
2. I know if there is just a ref that it's probably due to deletion but should we maybe call it something more vague like `UnknownEntityReceivedSnsMessage` on the off chance that there is some other reason for it reaching here?